### DISCUSSION TITLE
Resource Bundle Look-ups in Modular Java Applications - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#304] [Announcements]
# Resource Bundle Look-ups in Modular Java Applications - Gunnar Morling

The ResourceBundle class is Java’s workhorse for managing and retrieving locale specific resources,
such as error messages of internationalized applications.
With the advent of the module system in Java 9, specifics around discovering and loading resource bundles have changed quite a bit, in particular when it comes to retrieving resource bundles across the boundaries of named modules.

In this blog post I’d like to discuss how resource bundles can be used in a multi-module application
(i.e. a "modular monolith") for internationalizing error messages.
The following requirements should be satisified:

https://www.morling.dev/blog/resource-bundle-lookups-in-modular-java-applications/

**Comment by jvissers** [comment]
Nice, something I'll have ony list to check out early 2023!

**Reply by jvissers** [reply]
... on my ...

**Reply by gunnarmorling** [reply]
Hey @jvissers, thx for commenting! Do you have any specific use cases in mind? Moving Flink to the module system? FWIW, I haven't seen a huge appetite in the Java community around modules, unfortunately. In any case, a great 2023 to you!

**Comment by clydegerber** [comment]
Hi Gunnar! Thanks for posting this -- I have a couple of observations at https://javai18n.dev/guides/modular-resource-bundle-routing/.
